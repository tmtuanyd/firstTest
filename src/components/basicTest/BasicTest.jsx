import React from 'react';

const BasicTest = () => {
    return (
        <div>
            <h1>Basic test</h1>
            {/* <p>This is a basic test component</p>w */}
        </div>
    );
};

export default BasicTest;