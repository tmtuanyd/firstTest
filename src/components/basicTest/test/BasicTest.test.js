import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import BasicTest from "../basicTest";

Enzyme.configure({ adapter: new Adapter() });

//Basic test with enzyme
it("renders correctly enzyme", () => {
  const wrapper = shallow(<BasicTest />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders correctly enzyme with mount", () => {
  const wrapper = mount(<BasicTest />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
