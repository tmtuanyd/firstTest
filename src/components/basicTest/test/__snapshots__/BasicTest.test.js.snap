// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<div>
  <h1>
    Basic test
  </h1>
</div>
`;

exports[`renders correctly enzyme with mount 1`] = `
<BasicTest>
  <div>
    <h1>
      Basic test
    </h1>
  </div>
</BasicTest>
`;
